rust/core-lib/src/edit_types.rs

```diff
@@ -36,8 +36,8 @@ pub(crate) enum ViewEvent {
     Gesture { line: u64, col: u64, ty: GestureType },
     GotoLine { line: u64 },
     Find { chars: String, case_sensitive: bool, regex: Option<bool> },
-    FindNext { wrap_around: Option<bool>, allow_same: Option<bool> },
-    FindPrevious { wrap_around: Option<bool> },
+    FindNext { wrap_around: Option<bool>, allow_same: Option<bool>, modify_selection: Option<SelectionModifier> },
```

Comment:
since `SelectionModifier` has a `None` variant, we can maybe avoid having it be an `Option` here. One of the annotations serde lets us add to individual fields is `#[serde(default)]`, which will use the type's implementation of the `Default` trait if the field is missing. Then you just need to implement `Default` for `SelectionModifier`, and you can lose the `Option` while still making the field optional on the client side.

Reply:
(we should probably also be doing this for `wrap_around`. Maybe it makes sense to make these changes together in a follow-on PR)

Reply:
Yeah, probably best in separate PR.